### PR TITLE
[testing] Re-enable sklearn test_common::test_pandas_column_name_consistency test

### PR DESCRIPTION
**Pull request by icfaust**
### Description 
Check and fix issue with feature_name_in.

Fixes # - _issue number(s) if exists_

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [x] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Comment by icfaust** [comment]
Still needs to fix on Lasso and ElasticNet.  The solution is to follow the guideline set for Ridge.  The pandas data needs to be preserved in sending it to the sklearn fit method.

File: daal4py/sklearn/linear_model/_coordinate_descent.py
Comment (ethanglaser) [inline daal4py/sklearn/linear_model/_coordinate_descent.py:1]: not sure on purpose of these changes - there is a self._X and self._y in a function above - is it related? If its just because of naming conventions then there are probably a lot of places this occurs and maybe would be best for separate PR

File: daal4py/sklearn/linear_model/_ridge.py [hunk 1]
```diff
@@ -101,13 +101,11 @@ def _daal4py_predict(self, X):
     return res
 
 
-def _fit_ridge(self, X, y, sample_weight=None):
+def _fit_ridge(self, _X, _y, sample_weight=None):
     if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
         self._normalize = _deprecate_normalize(
             self.normalize, default=False, estimator_name=self.__class__.__name__
         )
-    if sklearn_check_version("1.0"):
```
Comment (ethanglaser) [inline daal4py/sklearn/linear_model/_ridge.py:109]: why remove?

File: sklearnex/linear_model/incremental_linear.py [hunk 2]
```diff
@@ -150,30 +151,32 @@ def _onedal_predict(self, X, queue=None):
...
-                dtype=[np.float64, np.float32],
-                copy=self.copy_X,
-            )
-            y = check_array(
-                y,
-                dtype=[np.float64, np.float32],
-                copy=False,
-                ensure_2d=False,
-            )
+        if not hasattr(self, "batch_size_"):
```
Comment (ethanglaser) [inline sklearnex/linear_model/incremental_linear.py:154]: why is condition needed?

**Review by ethanglaser — COMMENTED** [review]
Overall looks pretty good to me just need a few clarifications

File: daal4py/sklearn/linear_model/_ridge.py (hunk 1 above)
Comment (icfaust) [reply daal4py/sklearn/linear_model/_ridge.py:109]: check_features_names is an aspect which is called within sklearn's validate_data (not sure why they did it this way in the first place) https://github.com/scikit-learn/scikit-learn/blob/1.0.X/sklearn/base.py#L552. The inclusion of validate_data later on makes the whole if statement useless (and is in a sklearn_version_check("1.0"))

File: sklearnex/linear_model/incremental_linear.py (hunk 2 above)
Comment (icfaust) [reply sklearnex/linear_model/incremental_linear.py:154]: yeah, good catch.  batch_size is the only way to tell that onedal_partial_fit was called via `fit` and not `partial_fit`. If it was called from `partial_fit` data validation needs to occur on the input, if it was called from `fit` it has already been done.

Comment (icfaust) [reply sklearnex/linear_model/incremental_linear.py]: I changed it to match the convention in sklearn's IncrementalPCA which uses a check_input kwarg.

File: daal4py/sklearn/linear_model/_coordinate_descent.py
Comment (icfaust) [reply daal4py/sklearn/linear_model/_coordinate_descent.py:1]: Good question, I could use a different variable name.  Its purely to preserve the pandas dataframe in the case that the patchingConditionChain fails.  It definitely is confusing, maybe I will switch it given your concerns.

File: daal4py/sklearn/linear_model/_coordinate_descent.py [hunk 3]
```diff
@@ -435,9 +435,9 @@ def _daal4py_predict_lasso(self, X):
     return res
 
 
-def _fit(self, X, y, sample_weight=None, check_input=True):
+def _fit(self, _X, _y, sample_weight=None, check_input=True):
```
Comment (samir-nasibli) [inline daal4py/sklearn/linear_model/_coordinate_descent.py:438]: Why is this renaming required?

File: sklearnex/ensemble/_forest.py [hunk 4]
```diff
@@ -779,6 +778,7 @@ def _onedal_gpu_supported(self, method_name, *data):
                         or self.estimator.__class__ == DecisionTreeClassifier,
                         "ExtraTrees only supported starting from oneDAL version 2023.1",
                     ),
+                    (not self.oob_score, "oob_score value is not sklearn conformant."),
```
Comment (samir-nasibli) [inline sklearnex/ensemble/_forest.py:781]: Could you please bring more details in description ?

File: daal4py/sklearn/linear_model/_coordinate_descent.py (hunk 3 above)
Comment (icfaust) [reply daal4py/sklearn/linear_model/_coordinate_descent.py:438]: https://github.com/intel/scikit-learn-intelex/pull/1848#discussion_r1630084009

File: daal4py/sklearn/linear_model/_coordinate_descent.py
Comment (ethanglaser) [reply daal4py/sklearn/linear_model/_coordinate_descent.py:1]: I think I understand it now, _X and _y are the raw inputs and make sense to use them if falling back

**Review by ethanglaser — APPROVED** [review]
No objections, I generally find all of the X/y validation/pre-processing quite confusing so can't say I reviewed that part super diligently but looks good.
